core/dbt/tests/fixtures/project.py

```diff
@@ -454,5 +454,9 @@ def project(
 
     yield project
 
-    project.drop_test_schema()
+    # Some commands will not have an installed adapter when running and will raise a KeyError here.  Just pass for now.
+    try:
+        project.drop_test_schema()
+    except KeyError:
```

Comment:
There was another case where if you run dbt with another target the adapter will be switched and raise an error here. @gshank I feel like a better solution might be
- keep track of all the adapters we used in the test and loop over all of them to drop the schema it was using

Reply:
The reason that I switched to using the most recent adapter was that if we use an adapter that's not the latest we have to mock.patch the providers in order to avoid errors running sql, and we have to do it in every place that we connect to the database (this would be the case for any "saved" adapter that's not the latest).  Any adapter object should be able to drop the schema (if one exists) and it has the normal permissions.   The problem in the previous test that we dropped was that it was hacking the permissions in a kind of stupid way. The problem here is that there is no adapter -- which I didn't realize would be the case for deps, but makes sense because we now don't load the profile.

The more complete solution would be to check that an adapter exists and if it doesn't load one, but that seemed like too much work for this particular pull request.

Reply:
Opened ticket #5041

Reply:
Note: normally this would be easy to fix by running run_dbt() at the end of the test, but because this test fails when loading dependencies, that won't work.

Reply:
Right, I always forgot the persmission part. Then just check existance of adatper sounds perfect!